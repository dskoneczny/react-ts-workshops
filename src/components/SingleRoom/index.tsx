import React from 'react';
import { Card, CardBody, CardTitle, Button, Col } from 'reactstrap';

interface SingleRoomsProps {
  id: number;
  name: string;
  likes: number;
  addLike: (id: number) => void;
}

class SingleRoom extends React.PureComponent<SingleRoomsProps> {
  public addLike = () => {
    this.props.addLike(this.props.id);
  };

  public render() {
    return (
      <Col md='4'>
        <Card>
          <CardBody>
            <CardTitle>{this.props.name}</CardTitle>
            <div>
              Likes: {this.props.likes}
            </div>
            <Button onClick={this.addLike}>Like</Button>
          </CardBody>
        </Card>
      </Col>
    );
  }
}

export default SingleRoom;
