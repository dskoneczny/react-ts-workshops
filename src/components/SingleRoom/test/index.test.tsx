import * as React from 'react';
import SingleRoom from '../index';
import {shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Single room component test', () => {
  it('should render single room component', () => {
    const props = {
      addLike: jest.fn(),
      deleteRoom: jest.fn(),
      id: 12,
      likes: 0,
      name: 'example room',
    };
    const result = shallow(<SingleRoom {...props}/>);
    console.log(toJson(result));
    expect(toJson(result)).toMatchSnapshot();
  });
  it('should render single room component', () => {
    const props = {
      addLike: jest.fn(),
      deleteRoom: jest.fn(),
      id: 12,
      likes: 0,
      name: 'example room',
    };
    const result = mount(<SingleRoom {...props}/>);
    console.log(toJson(result));
  });

  it('should render single room component', () => {
    const props = {
      addLike: jest.fn(),
      deleteRoom: jest.fn(),
      id: 12,
      likes: 0,
      name: 'example room',
    };
    const result = render(<SingleRoom {...props}/>);
    console.log(toJson(result));
  });
});
