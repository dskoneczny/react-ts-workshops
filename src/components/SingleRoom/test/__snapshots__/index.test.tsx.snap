// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single room component test should render single room component 1`] = `
<Col
  md="4"
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  <Card
    tag="div"
  >
    <CardBody
      tag="div"
    >
      <CardTitle
        tag="div"
      >
        example room
      </CardTitle>
      <div>
        Likes: 
        0
      </div>
      <Button
        color="secondary"
        onClick={[Function]}
        tag="button"
      >
        Usuń pokój
      </Button>
    </CardBody>
  </Card>
</Col>
`;
